scapy/arch/bpf/core.py

```diff
@@ -37,14 +38,21 @@ def get_if_raw_addr(ifname):
     """Returns the IPv4 address configured on 'ifname', packed with inet_pton."""  # noqa: E501
 
     # Get ifconfig output
-    try:
-        fd = os.popen("%s %s" % (conf.prog.ifconfig, ifname))
-    except OSError as msg:
-        warning("Failed to execute ifconfig: (%s)", msg)
+    subproc = subprocess.Popen(
+        [conf.prog.ifconfig, ifname],
+        close_fds=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+    )
+    stdout, stderr = subproc.communicate()
+    if subproc.returncode:
+        warning("Failed to execute ifconfig: (%s)", stderr.decode("ascii"))
```

Comment:
I'd rather decode as utf8. (Multiple occurences)

We also have a `plain_str` function (from `scapy.compat`) that handles cross-compatibility between Python 2&3 if you're unsure of what to use.

Reply:
I think `plain_str` is best.

Using UTF-8 decoding can throw decoding exceptions if it runs into one or more characters that cannot be translated into UTF-8, which is unlikely, but possible if the `ifconfig` program crashes.